mowgli_etl/pipeline/wdc/parsimonious_parser/wdc_parsimonious_dimension_parser.py

```diff
@@ -32,27 +37,49 @@ def parse(self, *, entry: WdcOffersCorpusEntry):
         return_flag = False
         __VISITOR = WdcParsimoniousNodeVisitor()
 
-        def __generate_dimensions():
-            retVal = WdcProductDimensions.from_dict(dataclasses.asdict(__VISITOR.dictionary))
-            __VISITOR.reset()
-            if retVal.width.value is None and retVal.length.value is None and retVal.depth.value is None and retVal.height.value is None:
+        def __generate_dimensions(source):
+            __VISITOR = WdcParsimoniousNodeVisitor()
+            __VISITOR.visit(source)
+            retVal = WdcProductDimensions.from_dict(
+                dataclasses.asdict(__VISITOR.dictionary)
+            )
+            if (
+                retVal.width is None
+                and retVal.length is None
+                and retVal.depth is None
+                and retVal.height is None
+                and retVal.weight is None
+                and retVal.power is None
+            ):
                 return None
             return retVal
 
-        if entry.KeyValuePairs is not None:
-            if "dimensions" in entry.KeyValuePairs.keys():
-                KeyValuePairs = self.__GRAMMAR.parse(entry.KeyValuePairs["dimensions"])
-                __VISITOR.visit(KeyValuePairs)
-                result = __generate_dimensions()
+        if entry.key_value_pairs is not None:
+            if "dimensions" in entry.key_value_pairs.keys():
```

Comment:
Don't need .keys() here, "x" in somedict is the same thing.

Reply:
This looks the same as below except for "dimensions" instead of "weight". Just do
```
for key in ("dimensions", "weight"):
    key_value_pairs = ... entry.key_value_pairs[key] ...
```
?